.github/workflows/zulip_emoji_awaiting_author.yaml

```
@@ -0,0 +1,33 @@
+on:
+  pull_request:
+    types: [labeled, unlabeled]
+jobs:
+  set_pr_emoji:
+    if: github.event.label.name == 'awaiting-author'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout mathlib repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # donwload the full repository
```

```suggestion
          sparse-checkout: |
            scripts/zulip_emoji_merge_delegate.py
```
Is the full history really needed?  I think that just python script should be enough.  (Cf `.github/workflows/maintainer_bors.yml`.)